include/ck_tile/ops/fmha/kernel/fmha_fwd_kernel.hpp

```diff
@@ -744,28 +748,31 @@ struct FmhaFwdKernel
             }
         }();
 
-        auto dropout = [&, i_nhead_ = i_nhead, i_batch_ = i_batch]() {
-            if constexpr(kHasDropout)
-            {
-                return BlockDropout{i_batch_,
-                                    i_nhead_,
-                                    kargs.num_head_q,
-                                    kargs.drop_seed,
-                                    kargs.drop_offset,
-                                    kargs.rp_undrop,
-                                    kargs.p_undrop_in_uint8_t,
-                                    kargs.is_store_randval};
-            }
-            else
-            {
-                return NullBlockDropout{};
-            };
-        }();
+        // dropout
+        float rp_undrop             = 1;
+        uint8_t p_undrop_in_uint8_t = std::numeric_limits<uint8_t>::max();
+        uint64_t drop_seed          = 0;
+        uint64_t drop_offset        = 0;
+
+        if constexpr(kHasDropout)
+        {
+            rp_undrop           = kargs.rp_undrop;
+            p_undrop_in_uint8_t = kargs.p_undrop_in_uint8_t;
+            drop_seed           = kargs.drop_seed;
+            drop_offset         = kargs.drop_offset;
+        }
+        FmhaDropout dropout(i_batch,
+                            i_nhead,
+                            kargs.num_head_q,
+                            drop_seed,
+                            drop_offset,
+                            rp_undrop,
+                            p_undrop_in_uint8_t);
```

Comment:
A question,  why we still construct a  `FmhaDropout` object when  kHasDropout could be `false`?

Reply:
Refactored.